rust/main/agents/relayer/src/server/messages/insert_messages.rs

```diff
@@ -1,16 +1,19 @@
+use std::str::FromStr;
+
 use axum::{extract::State, http::StatusCode, Json};
 use serde::{Deserialize, Serialize};
 
 use hyperlane_base::server::utils::{
     ServerErrorBody, ServerErrorResponse, ServerResult, ServerSuccessResponse,
 };
-use hyperlane_core::HyperlaneMessage;
+use hyperlane_core::{HyperlaneMessage, H256};
 
 use crate::server::messages::ServerState;
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Message {
     pub message: HyperlaneMessage,
+    pub message_id: Option<String>,
```

Comment:
Why do we need separate field with message id while we have it on `HyperlaneMessage`?